feat(ai): add pipelines optimization flags

rickstaa opened the pull request:
> [!WARNING]
> Do not merge before https://github.com/livepeer/ai-worker/pull/61.

**What does this pull request do? Explain your changes. (required)**
<!-- A clear and concise description of what this pull request does. -->

This commit adds a new `OptimizationFlags` field to the `aiModels` config so that users can forward optimization environment variables to the [ai-worker](git@github.com:livepeer/ai-worker.git) for more information see https://github.com/livepeer/ai-worker/pull/61.

**Specific updates (required)**
<!--- List out all significant updates your code introduces -->

- New `OptimizationFlags` field added to `aiModels` config.
- Flags forwaded to the [ai-worker](git@github.com:livepeer/ai-worker.git) Warm function.
- A warning is thrown when users try to use the `optimizationFlags` with non warm containers.

**How did you test each of these updates (required)**

<!-- A detailed description of how you tested your code changes. Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc. -->

1. Ensured the package builds.
2. Ensured tests.sh were succesfull.
3. Spung up off-chain broadcaster.
4. Spung up off-chain orchestrator with `SFAST` optimization flag in the models configuration file.

```bash
[
    {
        "pipeline": "image-to-video",
        "model_id": "stabilityai/stable-video-diffusion-img2vid-xt-1-1",
        "price_per_unit": 3390842,
        "warm": true,
        "optimization_flags": {
            "SFAST": "true",
            "SOME": true
        }
    }
]

```

6. Let the ImageToVideo container warmup and check the container logs to see that SFAST is used and the model is pre-traced.
7. Send a ImageToVideo request to the broadcaster and checked that it stilled worked.

**Does this pull request close any open issues?**

<!-- Fixes # -->

No.

**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] Read the [contribution guide](./CONTRIBUTING.md)
- [x] `make` runs successfully
- [ ] All tests in `./test.sh` pass - @rickstaa we still need to fix the tests.
- [x] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

rickstaa committed:
feat(ai): add pipelines optimization flags

This commit adds a new `OptimizationFlags` field to the `aiModels`
config so that users can forward optimization environment variables to
the [ai-worker](git@github.com:livepeer/ai-worker.git) for more
information see https://github.com/livepeer/ai-worker/pull/61.

rickstaa commented:
@yondonfu, is there a specific reason why the binary execution proceeds even when an AIModelConfig is not specified? I encountered this issue while trying to activate flags for containers that don't start warm, as I wanted to pass Optimization flags directly to the AIWorker constructor.

https://github.com/livepeer/go-livepeer/blob/1019d429eca743218fe133602b5f4344e4766308/cmd/livepeer/starter/starter.go#L534-L605

rickstaa commented:
Requires https://github.com/livepeer/ai-worker/pull/61.

yondonfu commented:
@rickstaa 

> is there a specific reason why the binary execution proceeds even when an AIModelConfig is not specified?

I think it would sense to require a config file for `-aiModels` now because the config file is not only used to warm models, but also for specifying prices and determining which models to advertise during capability discovery. Prior to using the config file for configuring prices and capability discovery, the config file was left as optional because the O would just try to execute any request given a model ID. But now, since models have prices and have to be explicitly advertised, the config file is necessary.

yondonfu commented on cmd/livepeer/starter/starter.go:
```diff
@@ -538,19 +539,27 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
+				// Show warning if people set OptimizationFlags but not Warm.
+				if len(config.OptimizationFlags) > 0 && !config.Warm {
+					once.Do(func() {
+						glog.Warningf("OptimizationFlags set for model %v but Warm is not set. OptimizationFlags are currently only used for warm containers.", config.ModelID)
```
Wouldn't you want to print this warning for each model that does not have `warm: true` and is also configured to use `optimization_flags`?

yondonfu replied:
Since the JSON fields actually set by users use snake case IMO it makes sense to use `optimization_flags` and `warm` here instead of `OptimizationFlags` and `Warm` since those are the fields that the user would be familiar with.

yondonfu reviewed (commented):
Noting for the thread that this will need a go mod update to use the latest version of `ai-worker` once it is merged.

rickstaa committed:
chore: update ai-worker to latest commit

This commit ensures that the https://github.com/livepeer/ai-worker
dependency is on the latest commit that includes the new optimization
flags feature (see https://github.com/livepeer/ai-worker/pull/61).

rickstaa commented:
> @rickstaa
> 
> > is there a specific reason why the binary execution proceeds even when an AIModelConfig is not specified?
> 
> I think it would sense to require a config file for `-aiModels` now because the config file is not only used to warm models, but also for specifying prices and determining which models to advertise during capability discovery. Prior to using the config file for configuring prices and capability discovery, the config file was left as optional because the O would just try to execute any request given a model ID. But now, since models have prices and have to be explicitly advertised, the config file is necessary.

Ah thanks for the explanation. Makes sense. I will change the behavoir!

rickstaa committed:
refactor: improve OptFlags logging

This commit ensures that the `optimzation flag not supported` warning
is shown for each model that is not loaded warm.
